ReportingServicesTools/Functions/CatalogItems/Write-RsCatalogItem.ps1

```diff
@@ -187,22 +204,69 @@ function Write-RsCatalogItem
                 #region Upload other stuff
                 else
                 {
-                    $bytes = [System.IO.File]::ReadAllBytes($EntirePath)
-                    $warnings = $null
-                    try
+                    if ($itemType -eq 'Resource')
                     {
-                        $Proxy.CreateCatalogItem($itemType, $itemName, $RsFolder, $Overwrite, $bytes, $null, [ref]$warnings) | Out-Null
-                        if ($warnings)
+                        $additionalProperties = New-Object System.Collections.Generic.List[$propertyDataType]
+
+                        $descriptionProperty = New-Object $propertyDataType
+                        $descriptionProperty.Name = 'MimeType'
+                        if ($item.Extension -eq ".png")
+                        {
+                            $descriptionProperty.Value = 'image/png'
+                        }
+                        else
+                        {
+                            $descriptionProperty.Value = 'image/jpeg'
+                        }
+
+                        $additionalProperties.Add($descriptionProperty)
+
+                        $bytes = [System.IO.File]::ReadAllBytes($EntirePath)
+                        $warnings = $null
+                        try
                         {
-                          foreach ($warn in $warnings)
-                          {
-                            Write-Warning $warn.Message
-                          }
+                            $Proxy.CreateCatalogItem($itemType, $item.Name, $RsFolder, $Overwrite, $bytes, $descriptionProperty, [ref]$warnings)
+                            if ($warnings)
+                            {
+                                foreach ($warn in $warnings)
+                                {
+                                    Write-Warning $warn.Message
+                                }
+                            }
+                        }
+                        catch
+                        {
+                            throw (New-Object System.Exception("Failed to create resource item $($item.FullName): $($_.Exception.Message)", $_.Exception))
                         }
                     }
-                    catch
+                    else
                     {
-                        throw (New-Object System.Exception("Failed to create catalog item: $($_.Exception.Message)", $_.Exception))
+
+                        $additionalProperties = New-Object System.Collections.Generic.List[$propertyDataType]
+
+                        $descriptionProperty = New-Object $propertyDataType
```

Comment:
The only thing that changes between additionalProperties between resource and non resources is the property Name

I recommend to move the creation of the list, the creation of the descriptionPropery and the addition to the list outside of this if else to avoid code duplication